plugins/lua/sort/places.lua

```diff
@@ -0,0 +1,121 @@
+local _ENV = mkmodule('plugins.sort.places')
+
+local sortoverlay = require('plugins.sort.sortoverlay')
+local widgets = require('gui.widgets')
+
+local info = df.global.game.main_interface.info
+local buildings = info.buildings
+
+local function get_default_zone_name(zone_type)
+    if zone_type == df.civzone_type.Dump then return 'Garbage dump' end
+    if zone_type == df.civzone_type.WaterSource then return 'Water source' end
+    if zone_type == df.civzone_type.SandCollection then return 'Sand' end
+    if zone_type == df.civzone_type.FishingArea then return 'Fishing' end
+    if zone_type == df.civzone_type.Pond then return 'Pit/pond' end
+    if zone_type == df.civzone_type.Pen then return 'Pen/pasture' end
+    if zone_type == df.civzone_type.ClayCollection then return 'Clay' end
+    if zone_type == df.civzone_type.AnimalTraining then return 'Animal training' end
+    if zone_type == df.civzone_type.PlantGathering then return 'Gather fruit' end
+    if zone_type == df.civzone_type.Dungeon then return 'Dungeon' end
+    if zone_type == df.civzone_type.MeetingHall then return 'Meeting area' end
+    if zone_type == df.civzone_type.Barracks then return 'Barracks' end
+    if zone_type == df.civzone_type.ArcheryRange then return 'Archery range' end
+    if zone_type == df.civzone_type.Office then return 'Office' end
+    if zone_type == df.civzone_type.DiningHall then return 'Dining hall' end
+    if zone_type == df.civzone_type.Dormitory then return 'Dormitory' end
+    if zone_type == df.civzone_type.Bedroom then return 'Bedroom' end
+    if zone_type == df.civzone_type.Tomb then return 'Tomb' end
+
+    return '' -- If zone_type is anything else it will not have a default name populated in the Zones page when left nameless
+end
+
+local function get_zone_search_key(zone)
+    local site = df.global.world.world_data.active_site[0]
+    local result = {}
+
+    -- allow zones to be searchable by their name
+    if zone.name == nil or zone.name == '' then
+        table.insert(result, get_default_zone_name(zone.type))
+    else
+        table.insert(result, zone.name)
+    end
+
+    -- allow zones w/ assignments to be searchable by their assigned unit
+    if zone.assigned_unit ~= nil then
+        table.insert(result, dfhack.TranslateName(zone.assigned_unit.name))
+        table.insert(result, dfhack.units.getReadableName(zone.assigned_unit))
```

Comment:
You only need the last name (in English) from TranslateName since everything else is provided by `getReadableName`

However, it would be even better to call `sortoverlay.get_unit_search_key(zone.assigned_unit)` instead (unless you *don't* want the unit profession in the search key).